### PR TITLE
[flutter_plugin_android_lifecycle] Add example notice like android_intent_example

### PR DESCRIPTION
**Pull request by includecmath**
## Description

*I did some automation tools for plugins, and find flutter_plugin_android_lifecycle example is whiteboard. 
At first I thought something went wrong, so I review the document and project, but this is not an exception.
I suggest adding notice like other platform function plugin e.g. android_intent_example show 'This plugin only works with Android'.*

NO_DART_NOTICE

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/flutter_plugin_android_lifecycle/example/lib/main.dart [hunk 1]
```diff
@@ -12,6 +12,9 @@ class MyApp extends StatelessWidget {
         appBar: AppBar(
           title: const Text('Sample flutter_plugin_android_lifecycle usage'),
         ),
+        body: Center(
+            child: Text(
+                'This plugin only provides Android Lifecycle API\n for other Android plugins. NO DART!')),
```
Comment (ened) [inline packages/flutter_plugin_android_lifecycle/example/lib/main.dart]: I think this is a good message, but would probably be confused about the "NO DART!" part.

Could you reword this a bit, or even point the developer to a page on the Wiki / website?

Comment (includecmath) [reply packages/flutter_plugin_android_lifecycle/example/lib/main.dart]: Ok I change the Text.

**Comment by includecmath** [comment]
@ened, can you please review this?

**Comment by includecmath** [comment]
Hi @ened, can you please look this PR.

**Pull request merged**